File: backend-java/src/main/java/org/egualpam/contexts/observable/order/application/ports/in/FindOrder.java
package org.egualpam.contexts.observable.order.application.ports.in;

import org.egualpam.contexts.observable.order.application.ports.out.OrderRepository;

public class FindOrder {

    private final OrderRepository repository;

    public FindOrder(OrderRepository repository) {
        this.repository = repository;
    }

    public OrderDto execute(FindOrderQuery query) {
        var orderId = query.orderId();
        var order = repository.find(orderId)
                .orElseThrow(() -> new RuntimeException("Order not found"));

        return new OrderDto(order.id().value());
    }
}

File: backend/src/main/kotlin/org/egualpam/contexts/observable/order/application/ports/in/FindOrder.kt
package org.egualpam.contexts.observable.order.application.ports.`in`

import org.egualpam.contexts.observable.order.application.ports.out.OrderRepository

class FindOrder(private val repository: OrderRepository) {
  fun execute(query: FindOrderQuery): OrderDto {
    val order =
        query.orderId.let { repository.find(it) } ?: throw RuntimeException("Order not found")
    return OrderDto(order.getId().value())
  }
}

data class FindOrderQuery(val orderId: String)
data class OrderDto(val id: String)
